src/daemon/database/actions.rs

```diff
@@ -519,37 +520,72 @@ fn db_mark_vault_as(
     db_path: &Path,
     vault_id: u32,
     status: VaultStatus,
+    updated_at: u32,
 ) -> Result<(), DatabaseError> {
     db_exec(db_path, |tx| {
         tx.execute(
-            "UPDATE vaults SET status = (?1), updated_at = strftime('%s','now') \
-             WHERE vaults.id = (?2)",
-            params![status as u32, vault_id,],
+            "UPDATE vaults SET status = (?1), updated_at = (?2) \
+             WHERE vaults.id = (?3)",
+            params![status as u32, updated_at, vault_id],
         )
         .map_err(|e| DatabaseError(format!("Updating vault to '{}': {}", status, e.to_string())))?;
 
         Ok(())
     })
 }
 
-pub fn db_mark_spent_unvault(db_path: &Path, vault_id: u32) -> Result<(), DatabaseError> {
-    db_mark_vault_as(&db_path, vault_id, VaultStatus::Spent)
+pub fn db_mark_spent_unvault(
+    db_path: &Path,
+    vault_id: u32,
+    updated_at: u32,
```

Comment:
For clarity, can you name the var `block_time` instead of `updated_at` here and everywhere we use an actual block timestamp?